### PR TITLE
Inductor cpp wrapper: fix codegen of FallbackKernel with kwargs

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #104575

Fix cpp wrapper failure on TorchBench model `hf_Reformer` with `randn`:
```
random_rotations = torch.randn(rotations_shape, device=vectors.device, dtype=vectors.dtype)
```

For cpp wrapper, when `kwargs` is not empty, for `OpOverloadPacket` kernel, we need to know the exact overload schema to handle the `kwargs` properly when calling the cpp kernel: including finding the correct order of the kwargs and getting the default value for optional args without provided value when calling the function (`layout` in the above case).

The current support in this PR is conservative and we'll extend the functionality in subsequent PRs.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/_inductor/codegen/wrapper.py [hunk 1]
```diff
@@ -1173,7 +1174,7 @@ def generate_profiler_mark_wrapper_call(self, stack):
             'RECORD_FUNCTION("inductor_wrapper_call", c10::ArrayRef<c10::IValue>({{}}));'
         )
 
-    def codegen_device(self, device):
+    def codegen_option(self, device):
```
Comment (jgong5) [inline torch/_inductor/codegen/wrapper.py:1177]: Can we just fix `codegen_device` without adding the new `codegen_option`?

File: torch/_inductor/utils.py [hunk 2]
```diff
@@ -1198,13 +1198,6 @@ def is_optional(arg):
     return "Optional" in str(arg.type)
```
Comment (jgong5) [inline torch/_inductor/utils.py:1198]: Make this function inner too?

**Review by jgong5 — APPROVED** [review]

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/utils.py (hunk 2 above)
Comment (chunyuan-w) [reply torch/_inductor/utils.py]: Fixed

File: torch/_inductor/utils.py [hunk 3]
```diff
@@ -1156,3 +1156,107 @@ def red_text(msg):
...
+
+def is_optional(arg):
+    return "Optional" in str(arg.type)
+
+
+def args_error_message(nargs, max_pos_args, min_args):
+    if min_args != max_pos_args:
+        return f"takes from {min_args} to {max_pos_args} positional arguments but {nargs} were given"
+    else:
+        return f"takes {max_pos_args} positional arguments but {nargs} were given"
```
[the remark this answers is not in the document]
Comment (chunyuan-w) [reply torch/_inductor/utils.py]: Fixed

File: torch/_inductor/codegen/wrapper.py (hunk 1 above)
Comment (chunyuan-w) [reply torch/_inductor/codegen/wrapper.py]: Refactored the code to reuse `codegen_device` in `codegen_option`.
I checked that both of them are needed.

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**